### PR TITLE
Update search result description for secret gists

### PR DESCRIPTION
**Pull request by gregdeane**
### Why:
The documentation is not clear regarding secret gists appearing in search results.

### What's being changed (if available, include any code snippets, screenshots, or gifs):
Added clarification regarding search results and secret gists. Secret gists do appear in the search results for the person who created them (if they are logged in).

#### A secret gist
Screenshot 2022-09-19 at 09 06 07

#### The secret gist above appears in search results when I am logged in.
Screenshot 2022-09-19 at 09 06 35

### Check off the following:

- [x] I have reviewed my changes in staging (look for the "Automatically generated comment" and click the links in the "Preview" column to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

**Comment by cmwilson21** [comment]
@gregdeane Thanks so much for opening a PR! I'll get this triaged for review :zap:

File: content/get-started/writing-on-github/editing-and-sharing-content-with-gists/creating-gists.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ Every gist is a Git repository, which means that it can be forked and cloned. {%
...
+Secret gists don't show up in {% data variables.gists.discover_url %} and are not searchable (exception: if you are logged in and have created a secret gist, it will appear in the search results). Secret gists aren't private. If you send the URL of a secret gist to {% ifversion ghae %}another enterprise member{% else %}a friend{% endif %}, they'll be able to see it. However, if {% ifversion ghae %}any other enterprise member{% else %}someone you don't know{% endif %} discovers the URL, they'll also be able to see your gist. If you need to keep your code away from prying eyes, you may want to [create a private repository](/articles/creating-a-new-repository) instead.
```
Comment (skedwards88) [inline content/get-started/writing-on-github/editing-and-sharing-content-with-gists/creating-gists.md:24]:
```suggestion
Secret gists don't show up in {% data variables.gists.discover_url %} and are not searchable unless you are logged in and are the author of the secret gist. Secret gists aren't private. If you send the URL of a secret gist to {% ifversion ghae %}another enterprise member{% else %}a friend{% endif %}, they'll be able to see it. However, if {% ifversion ghae %}any other enterprise member{% else %}someone you don't know{% endif %} discovers the URL, they'll also be able to see your gist. If you need to keep your code away from prying eyes, you may want to [create a private repository](/articles/creating-a-new-repository) instead.
```

**Review by skedwards88 — COMMENTED** [review]
Thank you for this addition! I have one minor wording suggestion. I'll go ahead and incorporate that then get this merged down for you.

**Review by skedwards88 — APPROVED** [review]

**Comment by gregdeane** [comment]
@skedwards88 Great update! Much better now. Thank you.

**Pull request merged**